api/agent/agent.go

```diff
@@ -229,7 +229,11 @@ func (a *agent) Submit(callI Call) error {
 
 	slot, err := a.getSlot(ctx, call) // find ram available / running
 	if err != nil {
-		a.stats.DequeueAndFail(callI.Model().AppName, callI.Model().Path)
+		if err == context.DeadlineExceeded {
```

Comment:
Looks good, just one comment, maybe it is worth of having dedicated method because it does look like a duplication, no?